sdk/core/azure-core/azure/core/tracing/_abstract_span.py

```diff
@@ -113,8 +105,9 @@ def add_attribute(self, key, value):
         :type value: str
         """
 
-    def set_http_attributes(self, request, response=None):
-        # type: (HttpRequest, Optional[HttpResponseType]) -> None
+    def set_http_attributes(
+        self, request: "HttpRequest", response: Optional["HttpResponseType"] = None
```

Comment:
I noticed we put a lot of types under TYPE_CHECKING and use forward refs in azure-core. Is this for performance? It seems our docs improve without the forward refs:

![image](https://user-images.githubusercontent.com/31998003/211927577-929c7040-1ba2-4e01-8875-87d5a60fd969.png)

vs

![image](https://user-images.githubusercontent.com/31998003/211927617-9719c70c-b60a-4ef7-821a-bb3d1480d3b4.png)

Reply:
Yes it's for perf, and I think the typing in this file should be fixed to use a protocol, and not the actual type, will address that in a next PR